### PR TITLE
feat(core): migrate instantsearch.js to instantsearch-core

### PR DESCRIPTION
**Pull request by Haroenv**
**Summary**

migrate instantsearch constructor, index widget, connectors, utils, and remaining types from .js to instantsearch-core.

**Result**

Everything should work as is, minus a couple deprecated changes removed here to make migration simpler (eg. __hitIndex)

BREAKING CHANGE: __hitIndex is no longer populated

### CODE REVIEW

File: packages/instantsearch.js/src/lib/routers/__tests__/history.test.ts
Comment (dhayab) [inline packages/instantsearch.js/src/lib/routers/__tests__/history.test.ts:1]: Hasn't this moved to core/**/historyRouter.test.ts ?

**Review by dhayab — COMMENTED** [review]
Awesome work! I left a few questions.

Also, should we add a `BREAKING CHANGE` footer on the squashing commit? We're accumulating a number of them and it will help us list them all when we will release the major version.

File: packages/instantsearch.js/src/connectors/autocomplete/connectAutocomplete.ts
[the remark this answers is not in the document]
Comment (Haroenv) [reply packages/instantsearch.js/src/connectors/autocomplete/connectAutocomplete.ts:1]: if I would have a different exports form the InstantSearch.js files, I'd need to change React InstantSearch in this PR too

File: packages/instantsearch-core/types/instantsearch.d.ts [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+     * Changes the way root levels of hierarchical facets have their count displayed.
+     *
+     * If `false` (by default), the count of the refined root level is updated to match the count of the actively refined parent level.
+     *
+     * If `true`, the count of the root level stays the same as the count of all children levels.
+     *
+     * @default false
+     */
+    persistHierarchicalRootCount?: boolean;
+  };
```
[the remark this answers is not in the document]
Comment (Haroenv) [reply packages/instantsearch-core/types/instantsearch.d.ts:123]: I'll add a major comment indeed good point

File: packages/instantsearch-core/src/lib/public/__tests__/getWidgetAttribute.test.ts [hunk 2]
```diff
@@ -61,18 +61,6 @@ describe('getWidgetAttribute', () => {
     ).toBe('test');
   });
 
-  it('reads the attribute from a panel', () => {
```
[the remark this answers is not in the document]
Comment (Haroenv) [reply packages/instantsearch-core/src/lib/public/__tests__/getWidgetAttribute.test.ts:64]: it didn't seem so important to me, panel isn't really special and an InstantSearch.js only concept.

File: packages/algolia-experiences/package.json [hunk 3]
```diff
@@ -10,7 +10,7 @@
   ],
   "dependencies": {
     "algoliasearch": "5.1.1",
-    "instantsearch.js": "4.75.3"
+    "instantsearch-core": "0.1.0"
```
Comment (dhayab) [inline packages/algolia-experiences/package.json:13]: We're importing all the widgets from instantsearch.js in src/widgets.ts, does this change work?

Comment (Haroenv) [reply packages/algolia-experiences/package.json:13]: 🤦 yes needs to be changed indeed, I missed that import

**Review by dhayab — APPROVED** [review]
🚀

**Pull request merged**